### PR TITLE
feat(table): Update multiple selection iconography

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #10123

## Summary
Updates the iconography used for the multiple selection affordance in Table.

Visual changes:
Before:
Screenshot 2024-08-21 at 9 09 43 AM
Screenshot 2024-08-21 at 9 09 54 AM
Screenshot 2024-08-21 at 9 09 38 AM

After:
Screenshot 2024-08-21 at 9 09 17 AM
Screenshot 2024-08-21 at 9 09 21 AM
Screenshot 2024-08-21 at 9 09 24 AM

### CODE REVIEW

File: packages/calcite-components/src/components/table-header/resources.ts [hunk 1]
```diff
@@ -13,3 +13,9 @@ export const CSS = {
   lastCell: "last-cell",
   staticCell: "static-cell",
 };
+
+export const ICONS = {
+  allSelected: "check-square-f",
```
Comment (driskull) [inline packages/calcite-components/src/components/table-header/resources.ts:18]: nitpick: native checkbox calls these `checked`, `indeterminate` and `unchecked`. Although what you have might be easier to understand. :)

**Review by driskull — CHANGES_REQUESTED** [review]
Code looks good! Can we add screenshot test(s) to cover these changes?

File: packages/calcite-components/src/components/table-header/resources.ts (hunk 1 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/table-header/resources.ts]: Updated for consistency

**Review by driskull — APPROVED** [review]
🌈 👍

**Review by ashetland — APPROVED** [review]
Visuals look good!

**Pull request merged**